src/rbus/rbus_subscriptions.c

```diff
@@ -740,7 +762,8 @@ void rbusSubscriptions_resubscribeElementCache(rbusHandle_t handle, rbusSubscrip
             RBUSLOG_INFO("resubscribing %s for %s", sub->eventName, sub->listener);
             rtListItem_GetNext(item, &next);
             rtList_RemoveItem(subscriptions->subList, item, NULL);/*remove before calling subscribeHandlerImpl to avoid dupes in cache file*/
-            err = subscribeHandlerImpl(handle, true, el, sub->eventName, sub->listener, sub->componentId, sub->interval, sub->duration, sub->filter);
+            subscriptionId = &sub->subscriptionId;
```

Comment:
We need to revisit the logic of Provider crash n recovery stuff.
Because, the very first data model registration may be subscribed very later & the subscriptionID is 25.
But when crash & recovered, this method will be called while registering and it will get subscriptionID as 1.

The consumer thinks that the subscriptionID is 25 but the provider is having it as 1.